File: stocksingle.py
print("Enter elements of array seperated by space :")
s = str(input())
l = s.split(" ")
n = len(l)
arr = []
for i in range(0,n):
    arr.append(int(l[i]))
cdiff=0
diff=0
min=arr[0]
for i in range(0,n):
    cdiff=arr[i]-min
    if(cdiff>diff):
        diff=cdiff
        sell = i
    if(arr[i]<min):
        min=arr[i]
        buy=i
print("Output is in 1 based indexing")
print("Stock is  bought on day "+str(buy+1)+" sold on day "+str(sell+1)+" with profit of "+str(diff))

File: stocksingle.java
import java.util.*;
class stocksingle
{
    public static void main(String args[])
    {
        int n;
        Scanner sc = new Scanner(System.in);
        String s;
        System.out.println("Enter array elements seperated by space");
        s = sc.nextLine();
        System.out.println("Output on basis of 1 base Indexing");
        String [] str = s.split(" ");
        n= str.length;
        int [] arr =new int[n];
        int buy=0;
        int sell=0;
        for(int i=0;i<n;i++)
        {
            arr[i]=Integer.valueOf(str[i]);
        }
        int cdiff=0,diff=0,min=arr[0];
        for(int i=0;i<n;i++)
        {
            cdiff=arr[i]-min;
            if(cdiff>diff)
            {
            diff=cdiff;
            sell = i;
            }
            if(arr[i]<min)
            {
            min=arr[i];
            buy=i;
            }
        }
        System.out.println("Stock is  bought on day "+(buy+1)+" sold on day "+(sell+1)+" with profit of "+(diff));
    }
}        